src/authorized_keys_d/authorized_keys_d.go

```diff
@@ -259,11 +261,32 @@ func (akd *SSHAuthorizedKeysDir) Disable(name string) error {
 	return ak.Disable()
 }
 
-// Add adds the supplied key at name.
+// validateKeys validates keys via ssh.ParseAuthorizedKey.
+func validateKeys(keys []byte) error {
+	_, _, _, rest, err := ssh.ParseAuthorizedKey(keys)
+	if err == nil && len(rest) > 0 {
+		err = validateKeys(rest)
```

Comment:
It's worse than that, ssh.ParseAuthorizedKey() will return the same "ssh: no key found" error for altogether invalid input, making it impossible to differentiate "no key" due to comments or emptiness from an actual parse error:

>package main
>
> import (
>        "fmt"
>        "golang.org/x/crypto/ssh"
>)
>
>func main() {
>        _, _, _, rest, err := ssh.ParseAuthorizedKey([]byte("junky jibberish"))
>        fmt.Printf("got rest: %d err: %v\n", len(rest), err)
>}

Results in:
> got rest: 0 err: ssh: no key found

And if you stick a line of jibberish in there, followed by a valid ssh key, it doesn't even error at all it just returns the ssh key:
>package main
>
>import (
>"fmt"
>"golang.org/x/crypto/ssh"
>)
>
>func main() {
>_, _, _, rest, err := ssh.ParseAuthorizedKey([]byte("invalid junk\nssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQDRY8bv2nDBwXxfzzmfJ9Rx5EFMvEEWA12NM1mA+8ruJzmggYDjhFmTDyP/2uWjLUGn+0+GnBuC5L/JWlituzOCjboFN4WUblVlKygOVVI6zJxRhFNaIPUP/mEMe1lW7jyS7gwknEZ2SaYZbB95Hc9Z6MNZFYYGotJ3+ndof6qakxMfH3QCYk8EuP36rRBbPAP6LzKhTOaWOGmQpLmIRujLxWnj24wWNimf7mg8t7nzY18rzo9q8Plse8NwsNiAmVaI4JZphjp+Mey9qV8D8vz/qtBIHTxFmEEe02hnvUyqkRe+YKVHZQ8YRnHhbqScUPXHJuqViGsgxTiX2MGIgtgr vc@localhost\n# more comment\n"))
>fmt.Printf("got rest: %d err: %v\n", len(rest), err)
>}

Results in:
> got rest: 15 err: <nil>

So this thing is basically useless as-is, for verification purposes.